src/main/java/Duke.java

```diff
@@ -1,10 +1,108 @@
+                            throw new DukeUnspecifiedNumberException("mark");
+                        }
+                        try {
+                            index = Integer.valueOf(splitted[1]);
+                            tasks.markAsDone(index);
+                        } catch (NumberFormatException err) {
+                            throw new DukeInvalidNumberFormatException("mark");
+                        }
+                    } else if (command.startsWith("unmark")) {
+                        String[] splitted = command.split("\\s+");
+                        Integer index;
+                        if (splitted.length == 1) {
+                            throw new DukeUnspecifiedNumberException("unmark");
+                        }
+                        try {
+                            index = Integer.valueOf(splitted[1]);
+                            tasks.markAsNotDone(index);
+                        } catch (NumberFormatException err) {
+                            throw new DukeInvalidNumberFormatException("unmark");
+                        }
+                    } else if (command.startsWith("todo")) {
+                        tasks.addTodo(command);
+                    } else if (command.startsWith("deadline")) {
+                        tasks.addDeadline(command);
+                    } else if (command.startsWith("event")) {
+                        tasks.addEvent(command);
+                    } else if (command.startsWith("delete")) {
+                        String[] splitted = command.split("\\s+");
+                        Integer index;
+                        if (splitted.length == 1) {
+                            throw new DukeUnspecifiedNumberException("delete");
+                        }
+                        try {
+                            index = Integer.valueOf(splitted[1]);
+                            tasks.deleteTask(index);
+                        } catch (NumberFormatException err) {
+                            throw new DukeInvalidNumberFormatException("delete");
+                        }
+                    } else {
+                        throw new DukeException("You have entered an invalid command");
+                    }
+                }
+            } catch (DukeException ex) {
+                drawLine();
+                indentMessage(ex.getMessage());
+                drawLine();
+            }
+        }
+    }
+
+    private static void greetingMessage() {
+        drawLine();
+        indentMessage("Hello! I'm Duke");
+        indentMessage("What can I do for you?");
+        drawLine();
+    }
+
+    private static void bye() {
+        isClosed = true;
+        drawLine();
+        indentMessage("Bye, Hope to see you again soon!");
+        drawLine();
+    }
+
+    private static void echo(String msg) {
+        drawLine();
+        indentMessage(msg);
+        drawLine();
+    }
+
+    public static void indentMessage(String msg) {
```

Comment:
Using a verb here might be better as well!

Reply:
Right, I'll take it into consideration. Thank you for your comment.